pallets/did/src/errors.rs

```diff
@@ -22,59 +22,93 @@ use crate::did_details::DidVerificationKeyRelationship;
 
 /// All the errors that can be generated when validating a DID operation.
 #[derive(Debug, Eq, PartialEq, TypeInfo)]
-pub enum DidError {
-	/// See [StorageError].
-	StorageError(StorageError),
-	/// See [SignatureError].
-	SignatureError(SignatureError),
-	/// See [InputError].
-	InputError(InputError),
+pub enum Error {
```

Comment:
That's the top-level Error type for DIDs, so calling this `Error` in the context of the did::error module kind of makes sense. I agree that this also violates the "no error suffix" rule since technically "Error" is a suffix of "Error", but this is really an edgecase. Naming it "Did" could make sense, but it feels counter-intuitive to me. Perhaps we should modify the no-error suffix rule somehow?